src/huggingface_hub/hf_api.py

```diff
@@ -1615,46 +1615,133 @@ def run_as_future(self, fn: Callable[..., R], *args, **kwargs) -> Future[R]:
         self._thread_pool
         return self._thread_pool.submit(fn, *args, **kwargs)
 
+
+    from huggingface_hub.utils._auth import _get_token_from_google_colab, _get_token_from_environment, _get_token_from_file
+
+    def _get_token_source(self, token: Optional[Union[bool, str]] = None) -> str:
+        """
+        Determine the source of the token being used.
+        
+        Args:
+            token: The token parameter passed to whoami
+            
+        Returns:
+            str: Source of the token ("parameter", "environment", "file", "colab", or "unknown")
+        """
+        if isinstance(token, str):
+            return "parameter"
+        
+        # Use the helper functions to check token sources
+        if _get_token_from_environment():
+            return "environment"
+        
+        if _get_token_from_file():
+            return "file"
+        
+        if _get_token_from_google_colab():
+            return "colab"
+        
+        return "unknown"
```

Comment:
This can be removed now that `_get_token_source` is not used anymore (all the logic is already contained in `whoami`)